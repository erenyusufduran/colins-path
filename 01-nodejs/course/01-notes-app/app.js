const chalk = require("chalk");
const yargs = require("yargs");
const getNotes = require("./notes.js");

// console.log(process.argv);

yargs.command({
  command: "add",
  describe: "Add a new note",
  builder: {
    title: {
      describe: "Note title",
      demandOption: true,
      type: "string",
    },
    body: {
      describe: "Note body",
      demandOption: true,
      type: "string",
    },
  },
  handler: function (argv) {
    console.log("Title:", argv.title);
    console.log("Body: ", argv.body);
  },
});

yargs.command({
  command: "remove",
  describe: "Remove a note",
  handler: function () {
    console.log("Removing the note!");
  },
});

yargs.command({
  command: "list",
  describe: "List the notes",
  handler: function () {
    console.log("Listing out all notes");
  },
});

yargs.command({
  command: "read",
  describe: "Read a note",
  handler: function () {
    console.log("Reading a note");
  },
});

yargs.parse();

console.log(yargs.argv);
