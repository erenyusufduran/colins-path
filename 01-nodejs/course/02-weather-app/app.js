const request = require("request");
require("dotenv").config();

const url = `http://api.weatherstack.com/current?access_key=${process.env.ACCESS_KEY}&query=Kayseri`;

request({ url, json: true }, (error, response) => {
  const res = response.body.current;
  console.log(
    `It is currently ${res.temperature} degrees out. There is a %${res.precip} chance of rain`
  );
});
