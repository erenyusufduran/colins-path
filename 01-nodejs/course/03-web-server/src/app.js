const path = require("path");
const express = require("express");
const hbs = require("hbs");

const publicDirectoryPath = path.join(__dirname, "../public");
const viewsPath = path.join(__dirname, "../templates/views");
const partialsPath = path.join(__dirname, "../templates/partials");

const app = express();

// Setup handlebars engine and views location.
app.set("view engine", "hbs");
app.set("views", viewsPath); // this is for maybe wanna make something different views path name like templates.
hbs.registerPartials(partialsPath);

// Setup static directory to serve.
app.use(express.static(publicDirectoryPath));

app.get("", (req, res) => {
  res.render("index", {
    title: "Weather App",
    name: "Eren Yusuf Duran",
  });
});

app.get("/about", (req, res) => {
  res.render("about", {
    title: "About Page",
    name: "Eren Yusuf Duran",
  });
});

app.get("/help", (req, res) => {
  res.render("help", {
    helpText: "This is some helpful text.",
    title: "Help",
    name: "Eren Yusuf Duran",
  });
});

app.get("/weather", (req, res) => {
  res.send({
    forecast: "It is snowing",
    location: "Philadelphia",
  });
});

app.listen(3000, () => {
  console.log("Server is up on port 3000.");
});
