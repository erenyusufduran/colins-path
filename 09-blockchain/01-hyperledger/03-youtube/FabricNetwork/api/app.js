"use strict";
const log4js = require("log4js");
const logger = log4js.getLogger("BasicNetwork");
const bodyParser = require("body-parser");
const http = require("http");
const util = require("util");
const express = require("express");
const { expressjwt: expJWT } = require("express-jwt");
const jwt = require("jsonwebtoken");
const bearerToken = require("express-bearer-token");
const cors = require("cors");
const constants = require("./config/constants.json");

const host = process.env.HOST || constants.host;
const port = process.env.PORT || constants.port;

const helper = require("./app/helper");
const invoke = require("./app/invoke");
const query = require("./app/query");
const qscc = require("./app/qscc");

const app = express();

app.options("*", cors());
app.use(cors());
app.use(bodyParser.json());
app.use(
  bodyParser.urlencoded({
    extended: false,
  })
);
// set secret variable
app.set("secret", "thisismysecret");
app.use(
  expJWT({
    secret: "thisismysecret",
    algorithms: ["HS256"],
  }).unless({
    path: ["/users", "/users/login", "/register"],
  })
);
app.use(bearerToken());

logger.level = "debug";

app.use(async (req, res, next) => {
  logger.debug("New req for %s", req.originalUrl);
  if (
    req.originalUrl.indexOf("/users") >= 0 ||
    req.originalUrl.indexOf("/users/login") >= 0 ||
    req.originalUrl.indexOf("/register") >= 0
  ) {
    return next();
  }
  const token = req.token;
  if (token) {
    const decoded = jwt.verify(token, "thisismysecret");
    req.username = decoded.username;
    req.orgName = decoded.orgName;
    logger.debug(util.format(`Decoded from JWT token. Username: ${decoded.username}, Orgname: ${decoded.orgName}`));
    next();
  } else {
    return res.send({
      success: false,
      message: "Failed to auth token. Make sure to include the token.",
    });
  }
});

const server = http.createServer(app).listen(port, () => console.log(`Server started on ${port}`));
logger.info("**************** Server Started *****************");
logger.info(`***************** http://${host}:${port} *******************`);
server.timeout = 240000;

function getErrorMessage(field) {
  return { success: false, message: field + " field is missing or invalid in the request" };
}

// Register and enroll user
app.post("/users", async (req, res) => {
  const username = req.body.username;
  const orgName = req.body.orgName;
  logger.debug("End point : /users");
  logger.debug("User name : " + username);
  logger.debug("Org name  : " + orgName);
  if (!username) {
    res.json(getErrorMessage("'username'"));
    return;
  }
  if (!orgName) {
    res.json(getErrorMessage("'orgName'"));
    return;
  }

  const token = jwt.sign(
    {
      exp: Math.floor(Date.now() / 1000) + parseInt(constants.jwt_expiretime),
      username: username,
      orgName: orgName,
    },
    app.get("secret")
  );
  const response = await helper.getRegisteredUser(username, orgName, true);
  logger.debug("-- returned from registering the username %s for organization %s", username, orgName);
  if (response && typeof response !== "string") {
    logger.debug("Successfully registered the username %s for organization %s", username, orgName);
    response.token = token;
    res.json(response);
  } else {
    logger.debug("Failed to register the username %s for organization %s with::%s", username, orgName, response);
    res.json({ success: false, message: response });
  }
});

// Register and enroll user and get secret
app.post("/register", async (req, res) => {
  const username = req.body.username;
  const orgName = req.body.orgName;
  logger.debug("End point : /users");
  logger.debug("User name : " + username);
  logger.debug("Org name  : " + orgName);
  if (!username) {
    res.json(getErrorMessage("'username'"));
    return;
  }
  if (!orgName) {
    res.json(getErrorMessage("'orgName'"));
    return;
  }

  const token = jwt.sign(
    {
      exp: Math.floor(Date.now() / 1000) + parseInt(constants.jwt_expiretime),
      username: username,
      orgName: orgName,
    },
    app.get("secret")
  );

  const response = await helper.registerAndGetSecret(username, orgName);
  logger.debug("-- returned from registering the username %s for organization %s", username, orgName);
  if (response && typeof response !== "string") {
    logger.debug("Successfully registered the username %s for organization %s", username, orgName);
    response.token = token;
    res.json(response);
  } else {
    logger.debug("Failed to register the username %s for organization %s with::%s", username, orgName, response);
    res.json({ success: false, message: response });
  }
});

// Login and get jwt
app.post("/users/login", async (req, res) => {
  const username = req.body.username;
  const orgName = req.body.orgName;
  logger.debug("End point : /users");
  logger.debug("User name : " + username);
  logger.debug("Org name  : " + orgName);
  if (!username) {
    res.json(getErrorMessage("'username'"));
    return;
  }
  if (!orgName) {
    res.json(getErrorMessage("'orgName'"));
    return;
  }

  const token = jwt.sign(
    {
      exp: Math.floor(Date.now() / 1000) + parseInt(constants.jwt_expiretime),
      username: username,
      orgName: orgName,
    },
    app.get("secret")
  );

  const isUserRegistered = await helper.isUserRegistered(username, orgName);
  if (isUserRegistered) {
    res.json({ success: true, message: { token } });
  } else {
    res.json({
      success: false,
      message: `User with username ${username} is not registered with ${orgName}. Please register first`,
    });
  }
});

// Invoke transaction on chaincode on target peers
app.post("/channels/:channelName/chaincodes/:chaincodeName", async (req, res) => {
  try {
    logger.debug("==================== INVOKE ON CHAINCODE ==================");
    const peers = req.body.peers;
    const chaincodeName = req.params.chaincodeName;
    const channelName = req.params.channelName;
    const fcn = req.body.fcn;
    const args = req.body.args;
    logger.debug("channelName  : " + channelName);
    logger.debug("chaincodeName : " + chaincodeName);
    logger.debug("fcn  : " + fcn);
    logger.debug("args  : " + args);
    if (!chaincodeName) {
      res.json(getErrorMessage("'chaincodeName'"));
      return;
    }
    if (!channelName) {
      res.json(getErrorMessage("'channelName'"));
      return;
    }
    if (!fcn) {
      res.json(getErrorMessage("'fcn'"));
      return;
    }
    if (!args) {
      res.json(getErrorMessage("'args'"));
      return;
    }

    const message = await invoke.invokeTransaction(channelName, chaincodeName, fcn, args, req.username, req.orgName);
    console.log(`message result is : ${message}`);
    res.send({ result: message, error: null, errorData: null });
  } catch (error) {
    res.send({ result: null, error: error.name, errorData: error.message });
  }
});

app.get("/channels/:channelName/chaincodes/:chaincodeName", async (req, res) => {
  try {
    logger.debug("==================== QUERY BY CHAINCODE ==================");

    const channelName = req.params.channelName;
    const chaincodeName = req.params.chaincodeName;
    console.log(`chaincode name is :${chaincodeName}`);
    let args = req.query.args;
    let fcn = req.query.fcn;
    let peer = req.query.peer;

    logger.debug("channelName : " + channelName);
    logger.debug("chaincodeName : " + chaincodeName);
    logger.debug("fcn : " + fcn);
    logger.debug("args : " + args);

    if (!chaincodeName) {
      res.json(getErrorMessage("'chaincodeName'"));
      return;
    }
    if (!channelName) {
      res.json(getErrorMessage("'channelName'"));
      return;
    }
    if (!fcn) {
      res.json(getErrorMessage("'fcn'"));
      return;
    }
    if (!args) {
      res.json(getErrorMessage("'args'"));
      return;
    }
    console.log("args==========", args);
    args = args.replace(/'/g, '"');
    args = JSON.parse(args);
    logger.debug(args);
    const message = await query.query(channelName, chaincodeName, args, fcn, req.username, req.orgName);
    res.send({ result: message, error: null, errorData: null });
  } catch (error) {
    res.send({ result: null, error: error.name, errorData: error.message });
  }
});
