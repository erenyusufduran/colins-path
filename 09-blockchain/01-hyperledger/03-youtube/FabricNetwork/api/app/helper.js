"use strict";

const { Wallets } = require("fabric-network");
const path = require("path");
const FabricCaServices = require("fabric-ca-client");
const fs = require("fs");
const FabricCAServices = require("fabric-ca-client");

const getCCP = async (org) => {
  let ccpPath;
  org === "Org1" ? (ccpPath = path.resolve(__dirname, "..", "config", "connection-org1.json")) : null;
  org === "Org2" ? (ccpPath = path.resolve(__dirname, "..", "config", "connection-org2.json")) : null;
  org === "Org3" ? (ccpPath = path.resolve(__dirname, "..", "config", "connection-org3.json")) : null;
  const ccpJSON = fs.readFileSync(ccpPath, "utf-8");
  const ccp = JSON.parse(ccpJSON);
  return ccp;
};

const getCaURL = async (org, ccp) => {
  let caURL;
  org === "Org1" ? (caURL = ccp.certificateAuthorities["ca.org1.example.com"].url) : null;
  org === "Org2" ? (caURL = ccp.certificateAuthorities["ca.org2.example.com"].url) : null;
  org === "Org3" ? (caURL = ccp.certificateAuthorities["ca.org3.example.com"].url) : null;
  return caURL;
};

const getWalletPath = async (org) => {
  let walletPath;
  org === "Org1" ? (walletPath = path.join(process.cwd(), "org1-wallet")) : null;
  org === "Org2" ? (walletPath = path.join(process.cwd(), "org2-wallet")) : null;
  org === "Org3" ? (walletPath = path.join(process.cwd(), "org3-wallet")) : null;
  return walletPath;
};

const getAffilliation = async (org) => {
  return org === "Org1" ? "org1.department1" : "org2.department1";
};

const getRegisteredUser = async (username, userOrg) => {
  let ccp = await getCCP(userOrg);
  const caURL = await getCaURL(userOrg, ccp);
  console.log("Ca URL is", caURL);
  const ca = new FabricCaServices(caURL);

  const walletPath = await getWalletPath(userOrg);
  const wallet = await Wallets.newFileSystemWallet(walletPath);
  console.log(`Wallet path: ${walletPath}`);

  const userIdentity = await wallet.get(username);
  if (userIdentity) {
    console.log(`An identity for the user ${username} already exists in the wallet`);
    return { success: true, message: username + " enrolled successfully" };
  }

  let adminIdentity = await wallet.get("admin");
  if (!adminIdentity) {
    console.log('An identity for the admin user "admin" does not exist in the wallet');
    await enrollAdmin(userOrg, ccp);
    adminIdentity = await wallet.get("admin");
    console.log("Admin Enrolled Successfully");
  }

  const provider = wallet.getProviderRegistry().getProvider(adminIdentity.type);
  const adminUser = await provider.getUserContext(adminIdentity, "admin");
  let secret;
  try {
    secret = await ca.register({
      affiliation: await getAffilliation(userOrg),
      enrollmentID: username,
      role: "client",
      adminUser,
    });
    console.log(`Secret for the user with username: ${username} -------> ${secret}`);
  } catch (error) {
    return error.message;
  }

  const enrollment = await ca.enroll({ enrollmentID: username, enrollmentSecret: secret });
  const x509Identity = {
    credentials: {
      certificate: enrollment.certificate,
      privateKey: enrollment.key.toBytes(),
    },
    mspId: `${userOrg}MSP`,
    type: "X.509",
  };
  await wallet.put(username, x509Identity);
  console.log(`Successfully registered and enrolled admin user ${username} and imported it into the wallet`);
  return { success: true, message: username + " enrolled Successfully" };
};

const isUserRegistered = async (username, userOrg) => {
  const walletPath = await getWalletPath(userOrg);
  const wallet = await Wallets.newFileSystemWallet(walletPath);
  console.log(`Wallet path: ${walletPath}`);

  const userIdentity = await wallet.get(username);
  if (userIdentity) {
    console.log(`An identity for the user ${username} exists in the wallet`);
    return true;
  }
  return false;
};

const getCaInfo = async (org, ccp) => {
  let caInfo;
  org === "Org1" ? (caInfo = ccp.certificateAuthorities["ca.org1.example.com"]) : null;
  org === "Org2" ? (caInfo = ccp.certificateAuthorities["ca.org2.example.com"]) : null;
  org === "Org3" ? (caInfo = ccp.certificateAuthorities["ca.org3.example.com"]) : null;
  return caInfo;
};

const getOrgMSP = (org) => {
  let orgMSP;
  org === "Org1" ? (orgMSP = "Org1MSP") : null;
  org === "Org2" ? (orgMSP = "Org2MSP") : null;
  org === "Org3" ? (orgMSP = "Org3MSP") : null;
  return orgMSP;
};

const enrollAdmin = async (org, ccp) => {
  console.log("Calling enroll admin method");
  try {
    const caInfo = await getCaInfo(org, ccp);
    const caTLSCACerts = caInfo.tlsCACerts.pem;
    const ca = new FabricCAServices(caInfo.url, { trustedRoots: caTLSCACerts, verify: false }, caInfo.caName);

    const walletPath = await getWalletPath(org);
    const wallet = await Wallets.newFileSystemWallet(walletPath);
    console.log(`Wallet path: ${walletPath}`);

    const identity = await wallet.get("admin");
    if (identity) {
      console.log('An identity for the admin user "admin" already exists in the wallet');
      return;
    }

    const enrollment = await ca.enroll({ enrollmentID: "admin", enrollmentSecret: "adminpw" });
    console.log("Enrollment object is: ", enrollment);
    const x509Identity = {
      credentials: {
        certificate: enrollment.certificate,
        privateKey: enrollment.key.toBytes(),
      },
      mspId: `${org}MSP`,
      type: "X.509",
    };
    await wallet.put("admin", x509Identity);
    console.log('Successfully enrolled admin user "admin" and imported it into the wallet');
    return;
  } catch (error) {
    console.error(`Failed to enroll admin user "admin": ${error}`);
  }
};

const registerAndGetSecret = async (username, userOrg) => {
  const ccp = await getCCP(userOrg);
  const caURL = await getCaURL(userOrg, ccp);
  const ca = new FabricCaServices(caURL);

  const walletPath = await getWalletPath(userOrg);
  const wallet = await Wallets.newFileSystemWallet(walletPath);
  console.log(`Wallet path: ${walletPath}`);

  const userIdentity = await wallet.get(username);
  if (userIdentity) {
    console.log(`An identity for the user ${username} already exists in the wallet`);
    return { success: true, message: username + " enrolled successfully" };
  }

  // build a user object for authentication with CA
  const provider = wallet.getProviderRegistry().getProvider(adminIdentity.type);
  const adminUser = await provider.getUserContext(adminIdentity, "admin");
  let secret;
  try {
    secret = await ca.register(
      { affiliation: await getAffilliation(userOrg), enrollmentID: username, role: "client" },
      adminUser
    );
    const enrollment = await ca.enroll({ enrollmentID: username, enrollmentSecret: secret });
    const orgMSPID = getOrgMSP(userOrg);
    const x509Identity = {
      credentials: { certificate: enrollment.certificate, privateKey: enrollment.key.toBytes() },
      mspId: orgMSPID,
      type: "X.509",
    };
    await wallet.put(username, x509Identity);
    return { success: true, message: username + " enrolled successfully", secret };
  } catch (error) {
    return error.message;
  }
};

module.exports = {
  getCCP,
  getWalletPath,
  getRegisteredUser,
  isUserRegistered,
  registerAndGetSecret,
};
