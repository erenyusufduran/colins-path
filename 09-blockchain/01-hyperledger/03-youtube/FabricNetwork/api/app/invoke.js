const { Gateway, Wallets } = require("fabric-network");
const helper = require("./helper");
const { blockListener, contractListener } = require("./listeners");

const invokeTransaction = async (channelName, chaincodeName, fcn, args, username, orgName, transientData = []) => {
  try {
    const ccp = await helper.getCCP(orgName);
    console.log("==================", channelName, chaincodeName, fcn, args, username, org_name);

    const walletPath = await helper.getWalletPath(orgName);
    const wallet = await Wallets.newFileSystemWallet(walletPath);
    console.log(`Wallet path: ${walletPath}`);

    let identity = await wallet.get(username);
    if (!identity) {
      console.log(`An identity for the user ${username} does not exist in the wallet, so registering user`);
      await helper.getRegisteredUser(username, org_name, true);
      identity = await wallet.get(username);
      console.log("Run the registerUser.js application before retrying");
      return;
    }

    const gateway = new Gateway();
    await gateway.connect(ccp, { wallet, identity: username, discovery: { enabled: true, asLocalhost: true } });
    const network = await gateway.getNetwork(channelName);
    const contract = network.getContract(chaincodeName);

    // await contract.addContractListener(contractListener)
    // await network.addBlockListener(blockListener)

    let result;

    switch (fcn) {
      case "createCar":
        result = await contract.submitTransaction(fcn, args[0]);
        result = { txid: result.toString() };
        break;
      default:
        break;
    }

    gateway.disconnect();
    return result;
  } catch (error) {
    console.log(`Getting error: ${error}`);
    return error.message;
  }
};

exports.invokeTransaction = invokeTransaction;
