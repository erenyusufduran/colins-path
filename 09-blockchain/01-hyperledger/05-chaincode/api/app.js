"use strict";
const log4js = require("log4js");
const logger = log4js.getLogger("BasicNetwork");
const bodyParser = require("body-parser");
const http = require("http");
const util = require("util");
const express = require("express");
const expressJWT = require("express-jwt");
const jwt = require("jsonwebtoken");
const bearerToken = require("express-bearer-token");
const cors = require("cors");
const constants = require("./config/constants.json");

const app = express();
