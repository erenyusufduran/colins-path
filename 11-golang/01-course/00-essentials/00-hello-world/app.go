package main

//

import "fmt"

// go run app.go

func main() {
	fmt.Print("Hello World")
}
