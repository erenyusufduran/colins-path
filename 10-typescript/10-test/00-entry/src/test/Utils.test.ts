import { getStringInfo, toUpperCase } from '../app/Utils';

describe('Utils test suite', () => {
  test('Should return uppercase of valid string', () => {
    // arrange:
    const sut = toUpperCase;
    const expected = 'ABC';
    // act:
    const actual = sut('abc');

    // assert:
    expect(actual).toBe(expected);
  });

  describe('toUpperCase examples', () => {
    it.each([
      { input: 'abc', expected: 'ABC' },
      { input: 'My-String', expected: 'MY-STRING' },
      { input: 'def', expected: 'DEF' },
      { input: 'eren', expected: 'EREN' },
    ])('$input toUpperCase should be $expected', ({ input, expected }) => {
      const actual = toUpperCase(input);
      expect(actual).toBe(expected);
    });
  });

  describe('getStringInfo for arg My-String should', () => {
    test('Return right length', () => {
      const actual = getStringInfo('My-String');
      expect(actual.characters).toHaveLength(9);
    });
    test('Return right lowercase', () => {
      const actual = getStringInfo('My-String');
      expect(actual.lowerCase).toBe('my-string');
    });
    test('Return right uppercase', () => {
      const actual = getStringInfo('My-String');
      expect(actual.upperCase).toBe('MY-STRING');
    });
    test('Return right characters', () => {
      const actual = getStringInfo('My-String');
      expect(actual.characters).toEqual(['M', 'y', '-', 'S', 't', 'r', 'i', 'n', 'g']);
      expect(actual.characters).toContain<string>('M');
      expect(actual.characters).toEqual(expect.arrayContaining(['S', 't', 'r', 'i', 'n', 'g', 'M', 'y', '-']));
    });
    test('Return right extra info', () => {
      const actual = getStringInfo('My-String');
      expect(actual.extraInfo).not.toBe(undefined);
      expect(actual.extraInfo).not.toBeUndefined();
      expect(actual.extraInfo).toBeDefined();
      expect(actual.extraInfo).toBeTruthy();
    });
  });
});
