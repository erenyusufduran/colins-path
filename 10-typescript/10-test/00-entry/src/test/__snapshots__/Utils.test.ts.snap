// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils test suite StringUtils tests Should throw error on invalid argument - arrow function: "Invalid argument!" 1`] = `"Invalid argument!"`;

exports[`Utils test suite StringUtils tests Should throw error on invalid argument - arrow function: Invalid argument 1`] = `"Invalid argument!"`;

exports[`Utils test suite StringUtils tests Should throw error on invalid argument - arrow function: Invalid argument! 1`] = `"Invalid argument!"`;

exports[`Utils test suite StringUtils tests Should throw error on invalid argument - function: Invalid argument! 1`] = `"Invalid argument!"`;
